### PR TITLE
Feature: Change assigned avatar sheet (kids-1549)

### PR DESCRIPTION
**Pull request by Daniela510**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new avatar selection bottom sheet for improved avatar selection.
	- Added functionality to select and assign avatars based on user input.
- **Improvements**
	- Enhanced avatar retrieval logic for better state management.
	- Updated UI elements for a smoother user experience in the signup process.
- **Bug Fixes**
	- Removed unnecessary methods to streamline avatar management.

### CODE REVIEW

**Review by TammiLion — APPROVED** [review]

File: lib/features/registration/widgets/avatar_selection_bottomsheet.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import 'package:givt_app/features/family/features/avatars/cubit/avatars_cubit.dart';
+import 'package:givt_app/features/family/features/avatars/widgets/avatar_item.dart';
+import 'package:givt_app/features/family/shared/design/components/actions/fun_button.dart';
+import 'package:givt_app/features/family/shared/design/components/overlays/fun_bottom_sheet.dart';
+import 'package:givt_app/l10n/l10n.dart';
+import 'package:givt_app/shared/models/analytics_event.dart';
+import 'package:go_router/go_router.dart';
+
+class AvatarSelectionBottomsheet extends StatelessWidget {
+  const AvatarSelectionBottomsheet({required this.id, super.key});
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/registration/widgets/avatar_selection_bottomsheet.dart:15]: no.. this is a nitpick

File: lib/features/registration/widgets/avatar_selection_bottomsheet.dart [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  static void show(BuildContext context, String id) {
+    showModalBottomSheet<void>(
+      context: context,
+      isScrollControlled: true,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(10),
+      ),
+      backgroundColor: Colors.white,
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/registration/widgets/avatar_selection_bottomsheet.dart:86]: no

File: lib/features/registration/widgets/avatar_selection_bottomsheet.dart [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                      filename: state.avatars[index].fileName,
+                      url: state.avatars[index].pictureURL,
+                      isSelected:
+                          state.avatars[index] == state.getAvatarByKey(id),
+                    );
+                  },
+                );
+              },
+            ),
+          ),
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/registration/widgets/avatar_selection_bottomsheet.dart:60]: no

File: lib/features/registration/widgets/avatar_selection_bottomsheet.dart [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import 'package:givt_app/core/enums/amplitude_events.dart';
+import 'package:givt_app/features/family/features/avatars/cubit/avatars_cubit.dart';
+import 'package:givt_app/features/family/features/avatars/widgets/avatar_item.dart';
+import 'package:givt_app/features/family/shared/design/components/actions/fun_button.dart';
+import 'package:givt_app/features/family/shared/design/components/overlays/fun_bottom_sheet.dart';
+import 'package:givt_app/l10n/l10n.dart';
+import 'package:givt_app/shared/models/analytics_event.dart';
+import 'package:go_router/go_router.dart';
+
+class AvatarSelectionBottomsheet extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/registration/widgets/avatar_selection_bottomsheet.dart:14]: same nitpick twice... really? bs

File: lib/features/registration/widgets/avatar_selection_bottomsheet.dart [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+          ),
+          const SizedBox(height: 24),
+          FunButton(
+            text: context.l10n.save,
+            onTap: () => context.pop(),
+            rightIcon: FontAwesomeIcons.arrowRight,
+            analyticsEvent: AnalyticsEvent(
+              AmplitudeEvents.avatarSaved,
+              parameters: {
+                'filename': cubit.state.getAvatarByKey(id).fileName,
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/registration/widgets/avatar_selection_bottomsheet.dart:69]: no, getAvatarByKey returns an empty string otherwise. unnecessary

**Pull request merged**